docker-entrypoint.sh

```diff
@@ -1,110 +1,129 @@
 #!/bin/sh
 
-INTEGRITY_CHECK=${HUMHUB_INTEGRITY_CHECK:-1}
-WAIT_FOR_DB=${HUMHUB_WAIT_FOR_DB:-1}
-SET_PJAX=${HUMHUB_SET_PJAX:-1}
-AUTOINSTALL=${HUMHUB_AUTO_INSTALL:-"false"}
-
-HUMHUB_DB_NAME=${HUMHUB_DB_NAME:-"humhub"}
-HUMHUB_DB_HOST=${HUMHUB_DB_HOST:-"db"}
-HUMHUB_NAME=${HUMHUB_NAME:-"HumHub"}
-HUMHUB_EMAIL=${HUMHUB_EMAIL:-"humhub@example.com"}
-HUMHUB_LANG=${HUMHUB_LANG:-"en-US"}
-HUMHUB_DEBUG=${HUMHUB_DEBUG:-"false"}
+INTEGRITY_CHECK="${HUMHUB_INTEGRITY_CHECK:-1}"
+WAIT_FOR_DB="${HUMHUB_WAIT_FOR_DB:-1}"
+SET_PJAX="${HUMHUB_SET_PJAX:-1}"
+AUTOINSTALL="${HUMHUB_AUTO_INSTALL:-false}"
 
-export NGINX_CLIENT_MAX_BODY_SIZE=${NGINX_CLIENT_MAX_BODY_SIZE:-10m}
-
-wait_for_db() {
-	if [ "$WAIT_FOR_DB" == "false" ]; then
-		return 0
-	fi
+HUMHUB_HOST="${HUMHUB_HOST:-localhost}"
+HUMHUB_PROTO="${HUMHUB_PROTO:-http}"
+export HUMHUB_WEB_ROOT="${HUMHUB_WEB_ROOT:-/var/www/localhost/htdocs}"
+export HUMHUB_SUB_DIR="${HUMHUB_SUB_DIR:-}"
 
-	until nc -z -v -w60 $HUMHUB_DB_HOST 3306; do
-		echo "Waiting for database connection..."
-		# wait for 5 seconds before check again
-		sleep 5
-	done
+export NGINX_CLIENT_MAX_BODY_SIZE=${NGINX_CLIENT_MAX_BODY_SIZE:-10m}
+HUMHUB_DB_NAME="${HUMHUB_DB_NAME:-humhub}"
+HUMHUB_DB_HOST="${HUMHUB_DB_HOST:-db}"
+
+HUMHUB_NAME="${HUMHUB_NAME:-HumHub}"
+HUMHUB_EMAIL="${HUMHUB_EMAIL:-humhub@example.com}"
+HUMHUB_LANG="${HUMHUB_LANG:-en-US}"
+
+HUMHUB_DEBUG="${HUMHUB_DEBUG:-false}"
+
+wait_for_db () {
+  if [ "$WAIT_FOR_DB" == "false" ]; then
+    echo "Not waiting for database connection..."
+    return 0
+  fi
+
+  until nc -z -v -w60 db 3306
+  do
+    echo "Waiting for database connection..."
+    # wait for 5 seconds before checking again
+    sleep 5
+  done
 }
 
 echo "=="
-if [ -f "/var/www/localhost/htdocs/protected/config/dynamic.php" ]; then
-	echo "Existing installation found!"
-
-	wait_for_db
-
-	INSTALL_VERSION=$(cat /var/www/localhost/htdocs/protected/config/.version)
-	SOURCE_VERSION=$(cat /usr/src/humhub/.version)
-	cd /var/www/localhost/htdocs/protected/
-	if [[ $INSTALL_VERSION != $SOURCE_VERSION ]]; then
-		echo "Updating from version $INSTALL_VERSION to $SOURCE_VERSION"
-		php yii migrate/up --includeModuleMigrations=1 --interactive=0
-		php yii search/rebuild
-		cp -v /usr/src/humhub/.version /var/www/localhost/htdocs/protected/config/.version
-	fi
+# Look for root index.php rather than dynamic.php, which could be persisted with config files only
+if [ -f "${HUMHUB_WEB_ROOT}${HUMHUB_SUB_DIR}/index.php" ]; then
```

Comment:
So, I changed this because, in the event that a user has a persisted config directory but not a persisted web root, the entrypoint script would detect the `dynamic.php` when most of the other source files would be missing. Looking for the root `index.php` would detect the lack of source files in this case - I did this several times while experimenting with different configurations: dropping the web-root volume but keeping the configs and counting on the entrypoint script to replace the source files. Since the entrypoint script copies all the source files in just before it carries out the installation, the time when there is no `dynamic.php` present is short. Because of this, I decided that the root `index.php` file to be a better indicator of an `installation` - which I'm taking to mean the presence of all of the required source files, not migrating the database, adding configuration settings, etc. This is another change that is mostly just for my own use-case so if that doesn't make sense to you, that's fine!